tests/test-ospec.js

```diff
@@ -305,6 +305,17 @@ o.spec("ospec", function() {
 					done()
 				})
 			})
+			o("hooks work as intended the third time (arrow function)", (done) => {
+				callAsync(function() {
+					var spy = o.spy()
+					spy(a)
+
+					o(a).equals(1)
+					o(b).equals(1)
+
+					done()
+				})
+			})
```

Comment:
Please wrap this in an `eval` with a `try`/`catch`, so it can be avoided on platforms that don't support it (like IE). We don't *currently* test against IE, but I'd like to fix that eventually.

Basically, something like this:

```js
try {
	eval(
		'o("hooks work as intended the third time (arrow function)", (done) => {' +
		'	callAsync(function() {' +
		'		var spy = o.spy()' +
		'		spy(a)' +
		'' +
		'		o(a).equals(1)' +
		'		o(b).equals(1)' +
		'' +
		'		done()' +
		'	})' +
		'})'
	)
} catch (e) {
	// ignore
}
```

Reply:
ah, I had no idea IE doesn't support arrow functions.

Reply:
IE11 supports [very little ES6](https://kangax.github.io/compat-table/es6/#ie11), and IE10 supports basically none. Based on that, the only ES6 syntax IE seems to support is `let`/`const`, and even that has a few caveats.

Reply:
What's your thoughts about using something `document.documentMode`, which is only supported by [IE](https://www.w3schools.com/jsref/prop_doc_documentmode.asp) and run that test only if `document.documentMode` is `undefined`?

Reply:
I'd rather just go based on "does it support arrow functions". It's more explicit, and if in the future I decide to test it on [JerryScript](https://jerryscript.net/) or an old version of Rhino or something, I don't have to update the test.